src/dymaptic.GeoBlazor.Core/Components/Geometries/Geometry.cs

```diff
@@ -42,6 +51,44 @@ public class Geometry : MapComponent
     /// </summary>
     public virtual string Type => default!;
 
+    /// <summary>
+    /// Javascript runtime
+    /// </summary>
+    [Inject]
+    public IJSRuntime? JsRuntime { get; set; }
+
+    /// <summary>
+    /// Task to hold JavaScript module that has the logic to handle geometry related functions
+    /// </summary>
+    Task<IJSObjectReference>? _geometryJSModule = null;
+
+    /// <summary>
+    /// Bool to determine if geometry JS module has been initialized
+    /// </summary>
+    bool ModuleInitialized { get; set; }
+
+    /// <inheritdoc />
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            _geometryJSModule = JsRuntime?.InvokeAsync<IJSObjectReference>("import", $"./_content/dymaptic.GeoBlazor.Core/js/geometry.js").AsTask();
```

Comment:
As you probably ran into, `OnAfterRender` in Blazor does _not_ mean that the ArcGIS map has rendered yet. Hence your check for `MapRendered` below. In fact, I'm not sure whether `View` is populated here either. You might have to wait for _not_ `firstRender` to hook this up, or try moving line 79 (`await base.OnAfterRenderAsync`) up, since that will do the registration of this component with parent components.